### PR TITLE
Fix initialisation of versions in the DB

### PR DESCRIPTION
**Pull request by artonge**
Broken after https://github.com/nextcloud/server/pull/36690

`$userFolder` was only set for `SharedStorage`
Also took the opportunity to fix some psalm warnings.

### CODE REVIEW

**Review by juliusknorr — APPROVED** [review]

File: apps/files_versions/lib/Versions/LegacyVersionsBackend.php [hunk 1]
```diff
@@ -66,17 +66,35 @@ public function useBackendForStorage(IStorage $storage): bool {
...
 				throw new NotFoundException("version file not found for share owner");
 			}
+		} else {
+			$userFolder = $this->rootFolder->getUserFolder($user->getUID());
+		}
+
+		$fileId = $file->getId();
+		if ($fileId === null) {
+			throw new NotFoundException("File not found ($fileId)");
 		}
```
Comment (come-nc) [inline apps/files_versions/lib/Versions/LegacyVersionsBackend.php:98]: Move before if/else and remove duplicate?

Comment (artonge) [reply apps/files_versions/lib/Versions/LegacyVersionsBackend.php:98]: No possible as the check is needed at both places. `$file` is redeclared and used inside the `if`.

**Review by come-nc — APPROVED** [review]

**Review by come-nc — APPROVED** [review]

**Pull request merged**